### PR TITLE
chore: add dev-install xtask

### PR DESCRIPTION
**Pull request by b5**
fixes `private` config in the `xtask` crate (should be `publish = false`).

Adds new `cargo xtask dev-install` that'll copy all binaries to `$HOME/.cargo/bin/`, which is useful while I'm working on `iroh start`.

Optionally run `cargo xtask dev-install --build` to run a build of all binaries first

### CODE REVIEW

File: xtask/Cargo.toml [hunk 1]
```diff
@@ -2,12 +2,14 @@
 name = "xtask"
 version = "0.1.0"
 edition = "2021"
-private = true
+publish = false
```
Comment (dignifiedquire) [inline xtask/Cargo.toml:5]: good catch

File: xtask/src/main.rs [hunk 2]
```diff
@@ -43,13 +49,28 @@ fn dist() -> Result<()> {
...
 
     Ok(())
 }
 
-fn dist_binary() -> Result<()> {
+fn dev_install(build: bool) -> Result<()> {
+    if build {
+        dist().unwrap();
+    }
+    let bins = vec!["iroh", "iroh-one", "iroh-gateway", "iroh-p2p", "iroh-store"];
```
Comment (dignifiedquire) [inline xtask/src/main.rs:62]: could be an array (just remove the `vec!`)

File: xtask/src/main.rs [hunk 3]
```diff
@@ -43,13 +49,28 @@ fn dist() -> Result<()> {
...
+    let home = dirs_next::home_dir().unwrap();
+    for bin in bins {
+        let from = project_root().join(format!("target/release/{}", bin));
+        let to = home.join(format!(".cargo/bin/{}", bin));
+        println!("copying {} to {}", bin, to.to_str().unwrap());
```
Comment (dignifiedquire) [inline xtask/src/main.rs:67]: no unwrap please, `to.display()` is the way to go

File: xtask/src/main.rs [hunk 4]
```diff
@@ -43,13 +49,28 @@ fn dist() -> Result<()> {
...
+        fs::copy(from, to).unwrap();
```
Comment (dignifiedquire) [inline xtask/src/main.rs:68]: no `unwrap`, use the `?`

File: xtask/src/main.rs [hunk 5]
```diff
@@ -64,12 +85,7 @@ fn dist_binary() -> Result<()> {
 
     fs::copy(&dst, dist_dir().join("iroh"))?;
 
-    if Command::new("strip")
-        .arg("--version")
-        .stdout(Stdio::null())
-        .status()
-        .is_ok()
-    {
+    if which::which("strip").is_ok() {
```
Comment (dignifiedquire) [inline xtask/src/main.rs:88]: I missed it, do we really need `strip`?

Comment (b5) [reply xtask/src/main.rs:88]: nah probably not. I'll remove

**Review by Arqu — APPROVED** [review]
Thx, this is neat

**Pull request merged**